scripts/ioc_deploy.py

```diff
@@ -323,9 +353,19 @@ def main_perms(args: CliArgs) -> int:
         user_text = input("Confirm target? yes/true or no/false\n")
         if not is_yes(user_text, error_on_empty=False):
             return ReturnCode.NO_CONFIRM
+
+    return _perms(deploy_dir=deploy_dir, allow_write=allow_write, dry_run=args.dry_run)
+
+
+def _perms(deploy_dir: str, allow_write: bool, dry_run: bool) -> int:
+    """
+    Helper function: re-usable user-facing core of main_perms.
+
+    For example, this is used in both main_perms and in main_rebuild.
+    """
```

Comment:
Should this try-except block just be incorporated into `set_permissions`?  That function has basically the same signature as this one.

Reply:
You're right, I'll refactor a bit.